app/lib/replay/Problems.ts

```diff
@@ -48,86 +55,241 @@ export interface BoltProblem extends BoltProblemDescription {
+    }));
+
+    return problems;
   } catch (error) {
-    console.error("Error fetching problems", error);
-    toast.error("Failed to fetch problems");
+    console.error('Error fetching problems', error);
+    toast.error('Failed to fetch problems');
+
     return [];
   }
 }
 
 export async function getProblem(problemId: string): Promise<BoltProblem | null> {
+  if (!shouldUseSupabase) {
+    return replayGetProblem(problemId);
+  }
+
   try {
-    const rv = await sendCommandDedicatedClient({
-      method: "Recording.globalExperimentalCommand",
-      params: {
-        name: "fetchBoltProblem",
-        params: { problemId },
-      },
-    });
-    console.log("FetchProblemRval", rv);
-    const problem = (rv as { rval: { problem: BoltProblem } }).rval.problem;
-    if ("prompt" in problem) {
-      // 2/11/2025: Update obsolete data format for older problems.
-      problem.repositoryContents = (problem as any).prompt.content;
-      delete problem.prompt;
+    const { data, error } = await getSupabase()
+      .from('problems')
+      .select(
+        `
+        *,
+        problem_comments (
+          *
+        )
+      `,
+      )
+      .eq('id', problemId)
+      .single();
+
+    if (error) {
+      throw error;
     }
-    return problem;
+
+    return convertSupabaseProblemToBolt(data);
   } catch (error) {
-    console.error("Error fetching problem", error);
-    toast.error("Failed to fetch problem");
+    console.error('Error fetching problem', error);
+    toast.error('Failed to fetch problem');
   }
+
   return null;
 }
 
 export async function submitProblem(problem: BoltProblemInput): Promise<string | null> {
+  if (!shouldUseSupabase) {
+    return replaySubmitProblem(problem);
+  }
+
   try {
-    const rv = await sendCommandDedicatedClient({
-      method: "Recording.globalExperimentalCommand",
-      params: {
-        name: "submitBoltProblem",
-        params: { problem },
-      },
-    });
-    console.log("SubmitProblemRval", rv);
-    return (rv as any).rval.problemId;
+    const supabaseProblem = convertBoltProblemDescriptionToSupabase(problem);
+
+    const { data, error } = await getSupabase().from('problems').insert(supabaseProblem).select().single();
+
+    if (error) {
+      throw error;
+    }
+
+    return data.id;
   } catch (error) {
-    console.error("Error submitting problem", error);
-    toast.error("Failed to submit problem");
+    console.error('Error submitting problem', error);
+    toast.error('Failed to submit problem');
+
     return null;
   }
 }
 
-export async function updateProblem(problemId: string, problem: BoltProblemInput | undefined) {
+export async function updateProblem(problemId: string, problem?: BoltProblemInput): Promise<void | undefined> {
+  if (!shouldUseSupabase) {
+    return replayUpdateProblem(problemId, problem);
+  }
+
   try {
     if (!getNutIsAdmin()) {
-      toast.error("Admin user required");
-      return;
+      toast.error('Admin user required');
```

Comment:
Why all the style changes?
Can we please make sure to run a linter so we can keep consistent styles?

Reply:
This change is pulling in lint fixes. Will create a separate branch with all the lint fixes.